### PR TITLE
update.go: refactor calculatePostConfigChangeAction() 

### PR DESCRIPTION
**Pull request by mkenigs**
Currently, post config change actions are calculated as follows:
    1) Reboot if there's a force file
    2) Reboot if any MachineConfig fields change except for passwd
    3) Calculate an action based on file changes
    
    Layered updates will need to calculate post config change actions solely
    based on files, as they will be diffing two images. 1) and 3) can be
    consolidated into a single function that is shared between layered and
    legacy updates. All MachineConfig changes (2) will result in file
    changes on disk, which will cause a reboot if actions are calculated as
    normal. This is correct for any change except for changes to passwd,
    which currently do not require any post config change action. In order
    to preserve this behavior, add a reboot file exception for ssh keys,
    which is the only field the MCO supports modifying using passwd.

    Add some test cases to check that changes to ssh keys result in
    postConfigChangeActionNone when appropriate

### CODE REVIEW

File: pkg/daemon/update.go
Comment (kikisdeliveryservice) [inline pkg/daemon/update.go:476]: can you be explicit as to why we dont have to consider them here for future readers

**Review by kikisdeliveryservice — COMMENTED** [review]
a few questions/comments

File: pkg/daemon/update.go
Comment (mkenigs) [reply pkg/daemon/update.go:66]: `!dn.os.IsFCOS()`, so yeah

File: pkg/daemon/update.go
Comment (mkenigs) [reply pkg/daemon/update.go:472]: It can pass through an error from `calculatePostConfigChangeActionFromFiles`

File: pkg/daemon/update_test.go
Comment (mkenigs) [reply pkg/daemon/update_test.go:608]: There's two down at the bottom which I think cover every case. Anything I still need to add?

Comment (mkenigs) [reply pkg/daemon/update_test.go:608]: I looked at `osrelease.go` and I don't think there's any values for RHEL. For RHCOS, I'm not sure what VariantID should be. I guess I'd need to look at what os-release contains on those OSes. I just used `OperatingSystem{}` since `IsFCOS` will return false and that's the if branch I was trying to test

Comment (kikisdeliveryservice) [reply pkg/daemon/update_test.go:608]: Matthew updated the tests and this is fixed.
For posterity as per https://github.com/openshift/os/commit/31f295e3362a6622749a64a6ff610b727560bda1 and his manual checking RHCOS doesn't have Variant ID

**Comment by kikisdeliveryservice** [comment]
some aws IAM issues (across PRs) reported in 4-dev

/test e2e-aws-upgrade
/test e2e-aws

File: pkg/daemon/update.go
Comment (kikisdeliveryservice) [reply pkg/daemon/update.go:478]: ah duh, basically if mc contains a change in pwd other than ssh it'll never get to this function bc the mc isnt reconcilble. :) not sure why it took me so long to remember this.

**Review by kikisdeliveryservice — APPROVED** [review]
Good job, Matthew. It's a nice refactor the function flow makes sense to me. And we got to learn about Variant IDs :laughing:

**Comment by mkenigs** [comment]
> Good job, Matthew. It's a nice refactor the function flow makes sense to me. And we got to learn about Variant IDs laughing

Thanks, that's nice to hear :) Thanks for the review

**Comment by kikisdeliveryservice** [comment]
this needs another review and lgtm

@jkyros @cheesesashimi can you PTAL

File: pkg/daemon/update.go
Comment (jkyros) [inline pkg/daemon/update.go:437]: This feels maybe little weird having the forcefile check inside `calculatePostConfigChangeActionFromFiles` but its' no worse than what we had. Like, we factored out the other criteria so this function would suppoooosedly be making a decision just on this list of files provided...but then we also sneak out to the host's disk to check for a file (which means this function can't run as a 'hypothetical' against a list of files, it would need access to a host disk. 

From a "find the largest unit of code that both layered and non-layered can use" standpoint this is good. 

Cool for now, not worth holding this up for that, just something to think about for later.

**Review by jkyros — APPROVED** [review]
This makes sense to me and lets us use `calculatePostConfigChangeActionFromFiles` for layering without breaking existing. 
/lgtm

File: pkg/daemon/update.go
Comment (mkenigs) [reply pkg/daemon/update.go:437]: Yeah that's a good point

**Comment by mkenigs** [comment]
@kikisdeliveryservice do we want to cherry pick this?

**Pull request merged**
